diagnostics/physics/sss_eval.py

```diff
@@ -14,62 +14,54 @@
-
-    sw = xarray.open_mfdataset([os.path.join(rcdir, 'swa_95A4_s00_10.nc'), os.path.join(rcdir, 'swa_A5B7_s00_10.nc')], decode_times=False)
-    weights = xarray.DataArray([2004-1995+1, 2017-2005+1], dims=['time'], coords={'time': sw.time})
-    sw = sw.weighted(weights).mean('time')
-
-    gom = xarray.open_mfdataset([os.path.join(rcdir, 'gom_95A4_s00_10.nc'), os.path.join(rcdir, 'gom_A5B7_s00_10.nc')], decode_times=False)
-    weights = xarray.DataArray([2004-1995+1, 2017-2005+1], dims=['time'], coords={'time': gom.time})
-    gom = gom.weighted(weights).mean('time')
-
-    nwa = xarray.open_mfdataset([os.path.join(rcdir, 'nwa_95A4_s00_10.nc'), os.path.join(rcdir, 'nwa_A5B7_s00_10.nc')], decode_times=False, combine='nested', compat='override')
-    weights = xarray.DataArray([2004-1995+1, 2017-2005+1], dims=['time'], coords={'time': nwa.time})
-    nwa = nwa.weighted(weights).mean('time')
+    # Verify that xh/yh are set as coordinates, then make sure model coordinates match grid data
+    model_grid = model_grid.assign_coords( {'xh':model_grid.xh, 'yh':model_grid.yh } )
+    model_ave = xarray.align(model_grid, model_ave,join='override')[1]
 
+    # Open regional climatology datasets and merge together.
     # Convert to a common grid to merge the datasets together.
     # concat each dataset onto a new dimension,
     # then average over the dimension.
     # with no overlap, only one region has data at any point, 
     # so mean ignores the nans from other regions and selects the available data.
-    regional_grid = {'lat': np.arange(10.05, 65.10, 0.1), 'lon': np.arange(-97.95, -39.90, 0.1)}
+    regional_grid = {'lat': np.arange( config['lat']['south'], config['lat']['north'], 0.1), 'lon': np.arange(config['lon']['west'], config['lon']['east'], 0.1)}
     # For plotting later
-    regional_gridc = {'lat': np.arange(10.00, 65.15, 0.1), 'lon': np.arange(-98.0, -39.85, 0.1)}
-    nwa_i = xesmf.Regridder(nwa, regional_grid, method='bilinear', unmapped_to_nan=True)(nwa.s_an.isel(depth=0))
-    sw_i = xesmf.Regridder(sw, regional_grid, method='bilinear', unmapped_to_nan=True)(sw.s_an.isel(depth=0))
-    gom_i = xesmf.Regridder(gom, regional_grid, method='bilinear', unmapped_to_nan=True)(gom.s_an.isel(depth=0))
-    combined = xarray.concat([nwa_i, sw_i, gom_i], dim='region').mean('region')
+    regional_gridc = {'lat': np.arange( config['lat']['south']-0.05, config['lat']['north']+0.05, 0.1), 'lon': np.arange(config['lon']['west']-0.05, config['lon']['east']-0.05, 0.1)}
+    combined = combine_regional_climatologies( config, regional_grid)
 
     # Now interpolate to the model grid to compare.
     regional_to_mod = xesmf.Regridder({'lat': combined.lat, 'lon': combined.lon}, model_grid[['geolon', 'geolat']].rename({'geolon': 'lon', 'geolat': 'lat'}), method='bilinear')
     regional_rg = regional_to_mod(combined + 273) 
     regional_rg = regional_rg.where(regional_rg > 0) - 273
     delta_regional = model_ave - regional_rg
```

Comment:
This wouldn't have to be in this PR, and I also I should have done this a while ago. We should be able to replace these gymnastics (adding 273, setting anything <=0 to nan because xesmf produces 0s instead of nans, and then subtracting 273 back out) with just `unmapped_to_nan=True`